application/gradle/integration-test.gradle

```diff
@@ -1,31 +1,40 @@
-// Assumes 'java', 'groovy' or 'scala' plugins have been applied before
+configurations {
+    integrationTestCompile.extendsFrom testCompile
+    integrationTestRuntime.extendsFrom testRuntime
+}
 
-// Add integration test source sets
 sourceSets {
-    integrationTest { sourceSet ->
-        ["java", "groovy", "scala", "resources"].each {
-            if (!sourceSet.hasProperty(it)) return
-            sourceSet."$it".srcDir file("src/integration-test/${it}")
+    integrationTest {
+        java {
+            compileClasspath += main.output + test.output
+            runtimeClasspath += main.output + test.output
+            srcDir file("${rootDir}/sbsk-integration-test/src/test/java")
         }
+        resources.srcDir file("${rootDir}/sbsk-integration-test/src/test/resources")
     }
 }
 
-// Setup dependencies for integration testing
-dependencies {
-    integrationTestCompile sourceSets.main.output
-    integrationTestCompile sourceSets.test.output
-    integrationTestCompile configurations.testCompile
-    integrationTestRuntime configurations.testRuntime
-}
-
-// Define integration test task
 task integrationTest(type: Test) {
     testClassesDir = sourceSets.integrationTest.output.classesDir
     classpath = sourceSets.integrationTest.runtimeClasspath
-}
+    outputs.upToDateWhen { false }
```

Comment:
Perhaps we can create a `watcher` gradle task (for unit tests), which will be caching the tests, in order to help us during development.